code/datums/components/surgery_initiator.dm

```diff
@@ -90,6 +90,8 @@
 			continue
 		if(user == target && !(surgery.surgery_flags & SURGERY_SELF_OPERABLE))
 			continue
+		if(!iscarbon(target) && surgery.mob_biotypes && !(surgery.mob_biotypes & target.mob_biotypes))
+			continue
```

Comment:
This `iscarbon` shouldn't be necessary, if a surgery is intended to only target carbons it should be set in its `target_mobtypes` var